voltdrop_direct/aws-iot-core/index.js

```diff
@@ -0,0 +1,236 @@
+  }
+
+  // Packet ID - 1 byte
+  const packetId = raw[0];
+  switch (packetId) {
+    case packetList.VD_DIRECT_VOLTAGE_PF:
+      result.data = {
+        // Average Phase Voltages - 2 bytes each
+        // 16-bit unsigned integers in network byte order (MSB/BE) with 10 integer and 6 fractional bits
+        voltageL1: raw.readUInt16BE(1) / 64.0,
+        voltageL2: raw.readUInt16BE(3) / 64.0,
+        voltageL3: raw.readUInt16BE(5) / 64.0,
+        // Phase Power Factors - 1 byte each, 8-bit signed integers as percentage
+        powerFactorL1: raw.readInt8(7),
+        powerFactorL2: raw.readInt8(8),
+        powerFactorL3: raw.readInt8(9),
+        // Capacitor Voltage Scalar - 1 byte
+        // 8-bit unsigned integer representing the capacitor voltage.
+        // (as if the integer range from 0-255 is scaled to between 0.0V and 5.0V)
+        capacitorVoltage: raw[10] * capacitorVoltageFactor,
+      };
+      if (result.data.capacitorVoltage < 3.4) {
+        result.warnings.push(
+          "Low capacitor voltage may reduce transmit interval."
+        );
+      }
+      break;
+    case packetList.VD_DIRECT_AMPERAGE:
+      let currentL1 = raw.readUInt16BE(1) / 16.0;
+      let currentL2 = raw.readUInt16BE(3) / 16.0;
+      let currentL3 = raw.readUInt16BE(5) / 16.0;
+      result.data = {
+        // Average Phase Current - 2 bytes each
+        // 16-bit unsigned integers in network byte order (MSB/BE) with 12 integer and 4 fractional bits
+        currentL1: currentL1,
+        currentL2: currentL2,
+        currentL3: currentL3,
+        // Maximum Phase Current - 1 byte each
+        // 8-bit unsigned integer with 3 integer and 5 fractional bits (expressed as percentage in addition to 100%)
+        maxCurrentL1: (raw[7] / 32.0 + 1.0) * currentL1,
+        maxCurrentL2: (raw[8] / 32.0 + 1.0) * currentL2,
+        maxCurrentL3: (raw[9] / 32.0 + 1.0) * currentL3,
+        // Temperature Scalar
+        // 8-bit unsigned integer representing the temperature.
+        // (as if the integer range from 0-255 is scaled to between -40C and 80C)
+        temperatureCelsius: raw[10] * temperatureCelsiusFactor - 40.0,
+      };
+      break;
+    case packetList.VD_DIRECT_ACT_ENERGY_CONF: // Intentional fall-through
+    case packetList.VD_DIRECT_ACT_ENERGY_UNCONF:
+      result.data = {
+        // Total Forward Active Energy - 8 bytes
+        // Sum of all phases active energy accumulated in Watt-Hours since last factory reset downlink.
+        // 64-bit unsigned integer in network byte order (MSB/BE)
+        //
+        // NOTE: Due to the limitations of Javascript and JSON this codec only uses 53 bits
+        // (max of standard number type) This still gives an effective range of 102,821 years
+        // at 10 Mega-Watts which is more than the Voltdrop is reasonably capable of measuring.
+        // Test and truncate the BigInt type into a normal number for JSON compatibility to the
+        // ES5 version of the codec.
```

Comment:
Can remove mention of the ES5 version, probably cutoff after "Test and truncate the BigInt type into a normal number for JSON compatibility"

Reply:
Will do